Index: DeviceMonitoringAPI/urls.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>from django.urls import path\r\n\r\nfrom .views import IndexView, base\r\n\r\nurlpatterns = [\r\n    path('', IndexView.as_view(), name='index'),\r\n    path('base', base),\r\n]
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/DeviceMonitoringAPI/urls.py b/DeviceMonitoringAPI/urls.py
--- a/DeviceMonitoringAPI/urls.py	(revision cc2f546581ec210a3c775d0ed506874455ad9923)
+++ b/DeviceMonitoringAPI/urls.py	(date 1748479869148)
@@ -1,8 +1,9 @@
 from django.urls import path
 
-from .views import IndexView, base
+from .views import IndexView, base, SystemStatusAPIView
 
 urlpatterns = [
     path('', IndexView.as_view(), name='index'),
     path('base', base),
-]
\ No newline at end of file
+    path('api/system-status/', SystemStatusAPIView.as_view(), name='system-status'),
+]
Index: DeviceMonitoringAPI/views.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>from django.shortcuts import render\r\n\r\n# Create your views here.\r\n\r\nfrom django.views.generic import TemplateView\r\n\r\nclass IndexView(TemplateView):\r\n    template_name = 'index.html'\r\n\r\ndef base(request):\r\n    return render(request, 'base.html')
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/DeviceMonitoringAPI/views.py b/DeviceMonitoringAPI/views.py
--- a/DeviceMonitoringAPI/views.py	(revision cc2f546581ec210a3c775d0ed506874455ad9923)
+++ b/DeviceMonitoringAPI/views.py	(date 1748479793786)
@@ -1,11 +1,27 @@
 from django.shortcuts import render
 
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+
+from .services.system_monitor import collect_system_data
+from .serializers import SystemStatusSerializer
+
 # Create your views here.
 
 from django.views.generic import TemplateView
 
+
 class IndexView(TemplateView):
     template_name = 'index.html'
 
+
 def base(request):
-    return render(request, 'base.html')
\ No newline at end of file
+    return render(request, 'base.html')
+
+
+class SystemStatusAPIView(APIView):
+    def get(self, request):
+        system_data = collect_system_data()
+        serealizer = SystemStatusSerializer(system_data)
+        return Response(serealizer.data, status=status.HTTP_200_OK)
